scripts/common/kubernetes.sh

```diff
@@ -252,18 +252,30 @@ function kubernetes_has_remotes() {
     return 1
 }
 
+# Fetch the load balancer endpoint from the cluster.
+function existing_kubernetes_api_address() {
+    kubectl get cm -n kube-system kurl-config -o jsonpath='{ .data.kubernetes_api_address }'
+}
+
+# During the upgrade user might change the load balancer endpoint or want to use EKCO internal load balancer. So, we
+# to be checking the api endpoint status on the existing api server endpoint as the new endpoint is only available after
+# finishing the upgrade.
 function kubernetes_api_address() {
-    local addr="$LOAD_BALANCER_ADDRESS"
-    local port="$LOAD_BALANCER_PORT"
+    if [ -n "$upgrading_kubernetes" ]; then
+        existing_kubernetes_api_address
```

Comment:
what is the purpose of calling existing_kubernetes_api_address function here ?

Reply:
@jala-dx  It will fetch the api server address from cluster instead of variables defined in install.sh script. So, That during the upgrade with ekco we don't fail the build by validating the api server health from the variables ( localhost:6444 ) which might be not present yet.